### ISSUE TITLE
Checkbox plugin: disabled node gets checked when its parent is checked.

### ISSUE THREAD

**Issue by jarey** [vakata/jstree#1678]
As the title tells, when a enabled parent is checkd, all their children are checked too, even the disabled ones.

I think that this behaviour isn't correct since you can't check a disabled node alone. After is checked as said, you can not unchek it.

The correct behaviour IMHO is to not perform the check operation over any disabled node in any case.

Screenshot showing the use-case:

![image](https://cloud.githubusercontent.com/assets/5025224/21588621/2e2b4b68-d0e9-11e6-9733-8bdcc1ce377d.png)

Thanks. 

PS:If any pull request is needed (in case the bug is accepted, any guidance is greatly appreaciated to let me contribute to fix this issue).

**Comment by vakata** [comment]
Yes, this will not change - there is a huge discussion about this, but I can't find it at the moment. Unfortunately there is no one true solution, for some (me included) the current behavior is correct, for others (like you) - the correct behavior would be not to check the node. For me that would mean leaving the parent node as partially selected, but some disagree that the parent should be checked, even though some of its children are not checked. 

Basically it is a huge mess with no simple correct solution. The current behavior will not change for v.3, and I will see if I can come up with something for v.4. Please share any ideas on how you think the UI should behave - should the parent node be checked or partially checked, etc.

**Comment by jarey** [comment]
Thanks for your kind reply.

I will try to come up with some suggestions, as our team already discussed some UI behaviour just in the line you just referenced. We now see you have put a lot of effort on thinking all passible scenarios (as new users of the lib we just thought on first place that it could be a little bug, but trying to solve it on our own we faced the problems you just wrote).

Thanks again.
